offline license utilization reporting (#25844)
hashicorp/nomad: website/content/api-docs/operator/utilization.mdx

---
layout: api
page_title: Utilization - Operator - HTTP API
description: |-
  The /operator/utilization endpoints provide tools for generating utilization reporting bundles for Nomad Enterprise.
---

# Operator Utilization HTTP API

The `/operator/utilization` endpoints provide tools for generating utilization
reporting bundles for Nomad Enterprise.

<EnterpriseAlert />

## Generate Nomad Enterprise Utilization Report Buindle

> This endpoint generates a utilization report. If no utilization snapshot has
> been recorded in the previous 24 hours, Nomad will record a utilization snapshot
> first.

Reviewer: ```suggestion
This endpoint generates a utilization report. If Nomad did not record a
utilization snapshot in the previous 24 hours, Nomad records a utilization
snapshot first.
```

| Method | Path                       | Produces           |
|--------|----------------------------|--------------------|
| `POST` | `/v1/operator/utilization` | `application/json` |

> The table below shows this endpoint's support for [blocking queries][] and

Reviewer: ```suggestion
This table shows this endpoint's support for [blocking queries][] and
```

[required ACLs][].

| Blocking Queries | ACL Required     |
|------------------|------------------|
| `NO`             | `operator:write` |

### Parameters

- `today` `(bool: false)` - Specifies to return a bundle that only includes
  utilization snapshots from the previous 24 hours.

### Sample Request

```shell-session
$ curl -XPOST \
    https://localhost:4646/v1/operator/utilization?today=true
```

### Sample Response

The `Bundle` field in the response body is a base64-encoded blob. The exact
format of this blog is not part of the Nomad API and is subject to change
between versions. The [`nomad operator utilization`][] command, which uses this

> API, will decode this to a human-readable file in the current working directory.

Reviewer: ```suggestion
API, decodes this to a human-readable file in the current working directory.
```
